pandas/_libs/lib.pyx

```diff
@@ -238,6 +264,7 @@ def is_scalar(val: object) -> bool:
 
     # Note: PyNumber_Check check includes Decimal, Fraction, numbers.Number
     return (PyNumber_Check(val)
+            or is_pyarrow_scalar(val)
```

Comment:
are there any sequence-like scalars that could accidentally get caught by the PySequence_Check check?

Reply:
Ah, yes, that's a good point. For example, a ListScalar has `__getitem__`, and I think that alone is already enough to pass `PySequence_Check`. 

I can move it before `PySequence_Check` then? 

I am wondering if we shouldn't remove getitem/len from the ListScalar et al. Having scalars that also behave as a sequence is just very annoying, and it's only there for a bit of convenience (although we will also have to handle this for python list objects). 
(for example in shapely 2.0 we have been making all the LineString and Polygon objects non-sequences, to avoid all issues of putting those in numpy arrays)

Reply:
I'd be OK with moving it to before the PySequence_Check call.  I'm guessing the perf difference is only a few nanoseconds?